packages/block-library/src/button/edit.native.js

```diff
@@ -446,19 +500,33 @@ class ButtonEdit extends Component {
 export default compose( [
 	withInstanceId,
 	withColors( 'backgroundColor', { textColor: 'color' } ),
-	withSelect( ( select ) => {
+	withSelect( ( select, { clientId } ) => {
 		const { isEditorSidebarOpened } = select( 'core/edit-post' );
-		const { getSelectedBlockClientId, getBlockParents } = select(
-			'core/block-editor'
-		);
+		const {
+			getSelectedBlockClientId,
+			getBlockCount,
+			getBlockRootClientId,
+		} = select( 'core/block-editor' );
 
+		const parentId = getBlockRootClientId( clientId );
 		const selectedId = getSelectedBlockClientId();
-		const hasParents = getBlockParents( selectedId ).length > 0;
+		const isParentSelected = parentId === selectedId;
+		const numOfButtons = getBlockCount( parentId );
 
 		return {
 			selectedId,
 			editorSidebarOpened: isEditorSidebarOpened(),
-			hasParents,
+			parentId,
+			numOfButtons,
+			isParentSelected,
+		};
+	} ),
+	withDispatch( ( dispatch, { parentId } ) => {
+		const { removeBlock } = dispatch( 'core/block-editor' );
+		return {
+			onDelete: () => {
+				removeBlock( parentId );
+			},
```

Comment:
I think we can skip to create this action because we already jave it in Buttons, we just need to pass it in `block.native.js`.

The only thing is I'm not sure if proposed option is better from architectural point of view. WDYT ?